### PR TITLE
Enable go module support for the project

### PR DESCRIPTION
**Pull request by sadmansakib**
through this series of commits i have enabled go module support by following official [docs](https://blog.golang.org/using-go-modules). All README files are subjected to further changes

### CODE REVIEW

**Comment by lggomez** [comment]
Thanks for making this initial work

I already made the request to activate travis on this repo and see if we can bring it up on this PR, altough we're also thinking about using github actions travis already provides a simple and known way to start a test matrix so there's that

File: go.mod
Comment (oxisto) [inline go.mod:3]: This should probably be set to 1.14 to be consistent with the matrix build, possibly even lower

File: rsa_pss.go
Comment (oxisto) [inline rsa_pss.go:18]: Please refrain from just blindly doing a `s/dgrijalva/jwt-go/golang-jwt/jwt`. Until we know if we can transfer the old repositories, links to issues still need to refer to the old repo.

File: rsa_pss_test.go
Comment (oxisto) [inline rsa_pss_test.go:109]: see above

**Review by oxisto — CHANGES_REQUESTED** [review]

**Comment by oxisto** [comment]
There is an additional issue wrt to the go modules path. If we use `github.com/golang-jwt/jwt` but follow the original release cycle, i.e. `v3.x`, then this will break things. So we either need to use `github.com/golang-jwt/jwt` or re-start the release cycle from `v1.0.0`

File: go.mod
Comment (sadmansakib) [reply go.mod]: done

File: rsa_pss.go
Comment (sadmansakib) [reply rsa_pss.go]: sorry i blindly changed those without knowing.

**Comment by sadmansakib** [comment]
> There is an additional issue wrt to the go modules path. If we use `github.com/golang-jwt/jwt` but follow the original release cycle, i.e. `v3.x`, then this will break things. So we either need to use `github.com/golang-jwt/jwt/v3` or re-start the release cycle from `v1.0.0`

Is there any discussion regarding whether this fork will start from v1.0.0 release cycle or inherit v3.x release cycle from dgrijalva/jwt-go

**Comment by mfridman** [comment]
> > There is an additional issue wrt to the go modules path. If we use `github.com/golang-jwt/jwt` but follow the original release cycle, i.e. `v3.x`, then this will break things. So we either need to use `github.com/golang-jwt/jwt/v3` or re-start the release cycle from `v1.0.0`
> 
> Is there any discussion regarding whether this fork will start from v1.0.0 release cycle or inherit v3.x release cycle from dgrijalva/jwt-go

I created another issue #5 here to track adding module support. But as previously mentioned, this all depends on the preference of the original maintainer ([see comment here](https://github.com/dgrijalva/jwt-go/issues/462#issuecomment-846423835)).

After what happened with go-chi (https://github.com/go-chi/chi/issues/561), going from pre-modules v4 to module support with v1 (reasoning explained here https://github.com/go-chi/chi/pull/560) and ultimately settling on /v5 .. I'm a bit reluctant to merge this until we know the preference of the upstream repo.

**Comment by oxisto** [comment]
@sadmansakib: Could you please restore the `.travis.yml` file? thank you! otherwise, LGTM, once all the maintainers agreed on the release cycle.

**Comment by sadmansakib** [comment]
> @sadmansakib: Could you please restore the `.travis.yml` file? thank you! otherwise, LGTM, once all the maintainers agreed on the release cycle.

I have reverted the commit and resolved the conflicts

File: .travis.yml
Comment (sadmansakib) [reply .travis.yml:8]: i see my bad

Comment (sadmansakib) [reply .travis.yml]: @oxisto is it okay now?

Comment (oxisto) [reply .travis.yml]: Looks good. You also removed `tip` as well, but that's ok, that was actually accidentally left over, so everything looks good.

**Review by oxisto — APPROVED** [review]

**Review by Waterdrips — APPROVED** [review]

**Review by mfridman — APPROVED** [review]
lgtm. Thank you for your contribution.

**Pull request merged**
